frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -641,55 +650,65 @@ export const insightLogic = kea<insightLogicType>({
                 clearTimeout(values.timeout)
             }
         },
-        saveInsight: async ({ setViewMode }) => {
+        saveInsight: async ({ redirectToViewMode }) => {
             const insightNumericId =
                 values.insight.id || (values.insight.short_id ? await getInsightId(values.insight.short_id) : undefined)
-            if (!insightNumericId) {
-                throw new Error('Can only save saved insights whose id is known.')
-            }
 
-            if (emptyFilters(values.insight.filters)) {
+            if (insightNumericId && emptyFilters(values.insight.filters)) {
                 const error = new Error('Will not override empty filters in saveInsight.')
                 Sentry.captureException(error, {
                     extra: { filters: JSON.stringify(values.insight.filters), insight: JSON.stringify(values.insight) },
                 })
                 throw error
             }
-            const savedInsight: InsightModel = await api.update(
-                `api/projects/${teamLogic.values.currentTeamId}/insights/${insightNumericId}`,
-                {
-                    ...values.insight,
-                    dashboard: values.insight.dashboard || values.sourceDashboardId,
-                    derived_name: summarizeInsightFilters(
-                        values.insight.filters || {},
-                        values.aggregationLabel,
-                        values.cohortsById,
-                        values.mathDefinitions
-                    ).slice(0, 400),
-                    saved: true,
-                } as InsightModel
-            )
+
+            const { name, description, favorited, filters, deleted, layouts, color, dashboard, tags } = values.insight
```

Comment:
The insight contains so many pieces of data we don't want to send back to the API, so allow-listing here.

Reply:
This makes sense BUT this rationale would be even more useful as a comment in the code, so I added it in